java/src/jmri/jmrit/display/Editor.java

```diff
@@ -3093,72 +3112,48 @@ public void addToPopUpMenu(NamedBean nb, JMenuItem item, int menu) {
-                            || sti.getTurnout(SlipTurnoutIcon.LOWEREAST) == nb || sti.getTurnout(SlipTurnoutIcon.LOWERWEST) == nb) {
-                        switch (menu) {
-                            case VIEWPOPUPONLY:
-                                pos.getPopupUtility().addViewPopUpMenu(item);
-                                break;
-                            case EDITPOPUPONLY:
-                                pos.getPopupUtility().addEditPopUpMenu(item);
-                                break;
-                            default:
-                                pos.getPopupUtility().addEditPopUpMenu(item);
-                                pos.getPopupUtility().addViewPopUpMenu(item);
-                        }
+                    if ( sti.getTurnout(SlipTurnoutIcon.EAST) == nb || sti.getTurnout(SlipTurnoutIcon.WEST) == nb
+                        || sti.getTurnout(SlipTurnoutIcon.LOWEREAST) == nb
+                        || sti.getTurnout(SlipTurnoutIcon.LOWERWEST) == nb) {
+                        addToPopUpMenu( pos, item, menu);
                         return;
                     }
                 }
-            } else if (pos instanceof MultiSensorIcon) {
-                if (pos.getPopupUtility() != null) {
-                    MultiSensorIcon msi = (MultiSensorIcon) pos;
-                    boolean match = false;
-                    for (int i = 0; i < msi.getNumEntries(); i++) {
-                        if (msi.getSensorName(i).equals(nb.getUserName())) {
-                            match = true;
-                            break;
-                        } else if (msi.getSensorName(i).equals(nb.getSystemName())) {
-                            match = true;
-                            break;
-                        }
-                    }
-                    if (match) {
-                        switch (menu) {
-                            case VIEWPOPUPONLY:
-                                pos.getPopupUtility().addViewPopUpMenu(item);
-                                break;
-                            case EDITPOPUPONLY:
-                                pos.getPopupUtility().addEditPopUpMenu(item);
-                                break;
-                            default:
-                                pos.getPopupUtility().addEditPopUpMenu(item);
-                                pos.getPopupUtility().addViewPopUpMenu(item);
-                        }
+            } else if ( pos instanceof MultiSensorIcon && pos.getPopupUtility() != null) {
+                MultiSensorIcon msi = (MultiSensorIcon) pos;
+                for (int i = 0; i < msi.getNumEntries(); i++) {
+                    if ( msi.getSensorName(i).equals(nb.getUserName())
+                        || msi.getSensorName(i).equals(nb.getSystemName())) {
+                        addToPopUpMenu( pos, item, menu);
                         return;
                     }
                 }
             }
```

Comment:
Are you sure you will not get duplicate menu items now? Is it guaranteed that a sensor will only occur once in a multi sensor icon? I haven't looked at the code for MultiSensorIcon so I don't know.

Reply:
Hi @danielb987 , thanks for checking this over.
I'm presuming the existing logic is correct, no functional changes intended here.
I've had a second look at refactoring this and got the same result pretty much, ie looking for a single condition to be true to add the JPopupMenu.